community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/LeafPlannerList.scala

```diff
@@ -51,7 +52,13 @@ case class LeafPlannerList(leafPlanners: LeafPlanner*) {
 
           case _ => false
         }
-        if (!satisfied) throw new IndexHintException(name, label, property, "No such index found.")
+        if (!satisfied) {
+          if (context.useErrorsOverWarnings) throw new IndexHintException(name, label, property, "No such index found.")
+          else {
+            context.notificationLogger.log(IndexHintUnfulfilledNotification)
+            context.ignoredIndexHints += hint
```

Comment:
Don't see the reason why building extra state here, can we simply remove the hint from the query graph immediately here?